stdlib/public/core/SIMDVectorTypes.swift.gyb

```diff
@@ -185,6 +187,34 @@ extension SIMD${n} where Scalar: BinaryFloatingPoint {
   }
 }
 
+extension SIMD${n} : AdditiveArithmetic where Scalar : FloatingPoint {}
+
+extension SIMD${n} : VectorNumeric where Scalar : FloatingPoint {}
+
+extension SIMD${n} : Differentiable where Scalar : Differentiable & BinaryFloatingPoint {
+  public typealias TangentVector = SIMD${n}
+  public typealias CotangentVector = SIMD${n}
+  public typealias AllDifferentiableVariables = SIMD${n}
+  public func tangentVector(from cotangent: CotangentVector) -> TangentVector {
+    return cotangent
+  }
+}
+
+extension SIMD${n}
+  where Scalar : Differentiable & BinaryFloatingPoint,
+        Scalar.CotangentVector : BinaryFloatingPoint {
+  public func _vjpSubscript(index: Int) ->
+  (Scalar, (Scalar.CotangentVector) -> CotangentVector)
+  {
+    return (self[index], { (v: Scalar.CotangentVector) in
```

Comment:
I don't think the type signature is needed here. Did the compiler suggest otherwise?

```suggestion
    return (self[index], { v in
```

Reply:
At some point it was asking me for it either here in the struct or in the `SIMD` protocol where it gave me an error along the lines of `cannot convert return expression of type '(Self, (_) -> _)' to return type '(Self, (Self.CotangentVector) -> (Self.Scalar, Self.CotangentVector))'`, but I think this ended up being fixed when I corrected the protocol refinements. So I actually can remove these all now!